ee/backend/app/src/main/kotlin/io/tolgee/ee/api/v2/controllers/SuggestionController.kt

```
@@ -127,13 +126,28 @@ class SuggestionController(
+    languageId: Long,
+  ) {
+    if (suggestion.author?.id == authenticationFacade.authenticatedUser.id) {
+      return
+    }
+    try {
+      securityService.checkLanguageSuggestionsManagePermission(projectId, listOf(languageId))
+    } catch (e: LanguageNotPermittedException) {
+      // LanguageNotPermittedException is a PermissionException subtype — must be caught first and
+      // propagated as-is (LANGUAGE_NOT_PERMITTED); the generic arm below rewrites the message.
+      throw e
+    } catch (e: PermissionException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)